testclient/mainwindow.cpp

```diff
@@ -194,37 +194,37 @@ void MainWindow::sendPBClicked()
 void MainWindow::homeAllPBClicked()
 {
     addSLog(tr("Home All"));
-    core->pushCommand(GCode::toCommand(GCode::G28));
+    core->home();
 }
 
 void MainWindow::homeXPBClicked()
 {
     addSLog(tr("Home X"));
-    core->pushCommand(GCode::toCommand(GCode::G28, QStringLiteral("X0")));
+    core->home(QStringLiteral("X"));
 }
 
 void MainWindow::homeYPBClicked()
 {
     addSLog(tr("Home Y"));
-    core->pushCommand(GCode::toCommand(GCode::G28, QStringLiteral("Y0")));
+    core->home(QStringLiteral("Y"));
 }
 
 void MainWindow::homeZPBClicked()
 {
     addSLog(tr("Home Z"));
-    core->pushCommand(GCode::toCommand(GCode::G28, QStringLiteral("Z0")));
+    core->home(QStringLiteral("Z"));
 }
 
 void MainWindow::bedTempPBClicked()
 {
     addSLog(GCode::toString(GCode::M140));
-    core->pushCommand(GCode::toCommand(GCode::M140, ui->bedTempSB->cleanText()));
+    core->setBedTemp(ui->bedTempSB->value());
 }
 
 void MainWindow::extTempPBClicked()
 {
     addSLog(GCode::toString(GCode::M104));
-    core->pushCommand(GCode::toCommand(GCode::M104, ui->extTempSelCB->currentText().at(9), ui->extTempSB->cleanText()));
+    core->setExtruderTemp(ui->extTempSB->value(), QString(ui->extTempSelCB->currentText().at(9)).toInt());
```

Comment:
Ok, now I getting what are you doing.
Use QComboBox::currentIndex() and not this.